tekton-proposal.md

```diff
@@ -0,0 +1,73 @@
+
+Please leave comments on this PR. The text of the proposal may change based on feedback, and the PR will retain the history of these edits.
+
+## Background
+
+Knative Build was conceived at the beginning of the Knative project alongside the other serverless-focused components (Serving and Eventing) to enable a "serverless development experience". Its goal was to enable users to specify their source to a Knative Service deployment and the steps to take to make it a container, and deploy that container, while interacting with a single custom resource.
+
+From the beginning, Build had bigger ambitions. Its [2018 roadmap](https://github.com/knative/build/blob/master/roadmap-2018.md) mentions features that aren't directly related to furthering serverless deployments (triggering, workflow, etc.). These aren't accelerators for serverless deployments, these are CI/CD features!
+
+This tension has existed in Build from the beginning: did it exist to accelerate the Knative serverless platform? Or is it supposed to be a full-featured CI/CD platform? How should Build prioritize features that don't have anything to do with the rest of Knative? These are not hypothetical questions, these are questions we've heard from users, operators and contributors.
+
+To further add confusion, in August 2019 Knative began the Pipelines effort in the knative/build-pipeline repo. This was an experimental ground-up design to build on the experience with the Build resource, and build real CI/CD workflow infrastructure running on Kubernetes. Its [2019 roadmap](https://github.com/knative/build-pipeline/blob/master/roadmap-2019.md) (the knative/build repo never had a 2019 roadmap) included all the CI/CD features from 2018 and more, and makes only [passing mention](https://github.com/knative/build-pipeline/blob/master/roadmap-2019.md#dont-break-serving) of its relationship with Serving.
+
+Around the same time, Knative Serving added duck-typing support, which meant any Kubernetes resource that satisfies its barebones interface could be used in a pre-deployment build, and all of Build, TaskRun and PipelineRun satisfy that interface and can be used.
+
+Since the inception of the build-pipeline repo, most active development has happened there (with contributions from Pivotal, CloudBees, Red Hat and others). At this point, Knative Build is largely stable, requiring less than a single full-time engineer effort for requested features, high-priority bug fixes, and scheduled releases. This is not necessarily a bad thing! With the CI/CD features moving to Pipelines, there just aren't many serverless-focused features left for Build to implement at this time, and it could largely be considered "done" until it gets new feature requests from Knative's users and operators.
```

Comment:
Change "this is not necessarily a bad thing" into "this is most likely a good thing". A critical part of Knative is fully stable. As a matter of fact should be bump it to 1.0  to showcase this stability ?

Reply:
Done.

I don't think we're quite ready to call Build "stable" yet, or mint a v1.0. Even if Build was ready for that, it's subject to the current co-versioning scheme with Serving and Eventing, and I don't think they're ready for v1.0 yet. The v1.0 question seems like a separate question which I'd like to avoid bringing into this proposal.